### ISSUE TITLE
Add Git Workflow Documentation to DEVELOPERS.md

### ISSUE THREAD

**Issue by ijemafe** [PoE-Overlay-Community/PoE-Overlay-Community-Fork#15]
Up to this point, we have merged pull requests into the master branch, cutting releases at significant milestones.

There is now a v0.7.0 release from master snapshotting the changes up to this point. There is also now a `dev` branch.

We should discuss what our pull request, merge, and release processes are for accepting changes:

- Merge PRs into `master` or `dev`? If `dev`, disallow commits to master without a `dev` merge and subsequent release?
- What is the release process (branches, tagging, builds)?

**Comment by kifers** [comment]
It might be useful to have a "release train" branch (`master` works for this, but I've seen a dedicated branch used as well - named `release_train` or similar).

When developers work on an issue or feature request, they open a PR to the release train.  When finished, the PR is merged to release train.  When we have a set of features we want to release, we cut a branch (usually named `rel-x.y.z`, depending on the upcoming version).  This is useful if there are bugs found and we need to release a patch from the stable release.  If bugs are found and fixed, the release branch is merged back down to master before the next release branch is cut (or if the next branch exists, we merge to the new rel branch, then merge the new rel branch back down to master).

This is just how I've worked for the past few years, so I figured  I'd throw out the suggestion.
